### PR TITLE
libhns: Optimize buffer calculating process

### PR DESCRIPTION
**Pull request by Li-Weihang**
Calculating process of WQE buffer is not clear enough to understand, so
optimize the code to improve the readability.

Changelog:
v3:
- Use align() directly instead of defining a new macro hr_hw_page_align().

v2:
- Remove meaningless judgment of count in #1 as Leon suggests.
- Rewrite commit message of both patch.

### CODE REVIEW

File: providers/hns/hns_roce_u.h [hunk 1]
```diff
@@ -263,6 +268,20 @@ struct hns_roce_u_hw {
 	struct verbs_context_ops hw_ops;
 };
 
+/*
+ * The entries's buffer should be aligned to a multiple of the hardware's
+ * minimum page size.
+ */
+#define hr_hw_page_align(x)		align(x, HNS_HW_PAGE_SIZE)
+
+static inline unsigned int to_hr_hem_entries_size(int count, int buf_shift)
```
Comment (rleon) [inline providers/hns/hns_roce_u.h:277]: This function doesn't make any sense.

if count == 0, hr_hw_page_align will return 0

File: providers/hns/hns_roce_u_verbs.c [hunk 2]
```diff
@@ -634,137 +618,153 @@ static int hns_roce_alloc_recv_inl_buf(struct ibv_qp_cap *cap,
...
-		qp->sq.wqe_cnt = align_qp_size(attr->cap.max_send_wr);
-		qp->rq.wqe_cnt = align_qp_size(attr->cap.max_recv_wr);
-	} else {
-		qp->sq.wqe_cnt = align_queue_size(attr->cap.max_send_wr);
-		qp->rq.wqe_cnt = align_queue_size(attr->cap.max_recv_wr);
-	}
+	/* RQ WQE */
+	qp->rq.max_gs = max(1U, attr->cap.max_recv_sge);
+	if (hr_dev->hw_version == HNS_ROCE_HW_VER1)
+		qp->rq.wqe_shift = hr_ilog32(sizeof(struct hns_roce_rc_rq_wqe));
```
Comment (rleon) [inline providers/hns/hns_roce_u_verbs.c:731]: I see that I missed the commit 3ebc62089441860367c0534fb40832a1c9d251b6.
All those hr_ilog32() should be converted to be like in this patch 69dbf7ccd18af7c7dd18a16779b03fbce59def45

File: providers/hns/hns_roce_u.h [hunk 3]
```diff
@@ -119,7 +119,7 @@ struct hns_roce_buf {
 };
 
 #define BIT_CNT_PER_BYTE       8
-#define BIT_CNT_PER_U64		64
+#define BIT_CNT_PER_LONG       (BIT_CNT_PER_BYTE * sizeof(unsigned long))
```
Comment (rleon) [inline providers/hns/hns_roce_u.h:122]: The change doesn't align with commit message.
Please explain why are you doing what are you doing.
Why did you change variable types from clean uint32_t/uin64_t to long?
Why did you mix them?

File: providers/hns/hns_roce_u.h (hunk 1 above)
Comment (Li-Weihang) [reply providers/hns/hns_roce_u.h:277]: I see, thanks for you comments. Will fix it.

File: providers/hns/hns_roce_u_verbs.c (hunk 2 above)
Comment (Li-Weihang) [reply providers/hns/hns_roce_u_verbs.c:731]: Thanks for your reminder, but I have fixed it in 3ebc620 :)

File: providers/hns/hns_roce_u.h (hunk 3 above)
Comment (Li-Weihang) [reply providers/hns/hns_roce_u.h:122]: This macro is for bitmap in srq, and it's defined in type of unsigned long, so I think PER_LONG is better than PER_U64. Is it ok?
Sorry for that I didn't mentioned it in the commit message, will modify it and pay more attention on the commit message.

File: providers/hns/hns_roce_u.h [hunk 4]
```diff
@@ -263,6 +268,17 @@ struct hns_roce_u_hw {
...
 
+/*
+ * The entries's buffer should be aligned to a multiple of the hardware's
+ * minimum page size.
+ */
+#define hr_hw_page_align(x)		align(x, HNS_HW_PAGE_SIZE)
+
+static inline unsigned int to_hr_hem_entries_size(int count, int buf_shift)
+{
+	return hr_hw_page_align(count << buf_shift);
```
Comment (jgunthorpe) [inline providers/hns/hns_roce_u.h:279]: The only place the macro hr_hw_page_align is used is here, just call align() directly

Comment (Li-Weihang) [reply providers/hns/hns_roce_u.h]: Thank you, I have fixed it.
